src/Homepage/index.js

```diff
@@ -66,65 +49,83 @@ class Homepage extends React.Component {
 
-    const queryParams = qs.parse(this.props.location.search, { ignoreQueryPrefix: true });
-    if ('submittedStore' in queryParams) {
+    const queryParams = qs.parse(this.props.location.search, {
+      ignoreQueryPrefix: true,
+    });
+    if ("submittedStore" in queryParams) {
       const storeData = JSON.parse(atob(queryParams.submittedStore));
-      selectedLocation = { lat: storeData.Latitude, lng: storeData.Longitude }
+      selectedLocation = { lat: storeData.Latitude, lng: storeData.Longitude };
       selectedStoreName = storeData[FORM_FIELDS.STORE_NAME];
     }
 
-    const newCenter = selectedLocation || location;
-    this.setState({
-      results: this.formatResults(data, newCenter),
-      markers: data.map(result => ({
-        lat: Number(result.Latitude),
-        lng: Number(result.Longitude),
-      })),
-      isLoading: false,
-      selectedLocation,
-      selectedStoreName,
-      searchResultLatlng: newCenter,
-      mapShouldPan: Boolean(selectedLocation),
-      showFormSubmissionNotification: Boolean(selectedLocation),
-    }, () => {
-      // remove query param from url
-      this.props.history.replace('/');
+    let queryLocation = selectedLocation || this.state.searchResultLatlng;
+    const response = await api.query({
+      ...queryLocation,
+      radius: DISTANCE_FILTER,
+    });
+    const { results: data, location: locationComingFromServer } = response;
+    if (!queryLocation) {
+      this.props.setIPlocation(locationComingFromServer);
+    }
+
+    const isLocationSelected = Boolean(selectedLocation);
+    const newCenter = selectedLocation || locationComingFromServer;
 
-      if (selectedLocation) {
-        setTimeout(() => this.setState({ mapShouldPan: false }), 1000);
-        const searchResultsContainer = document.querySelector('.search-results-container');
+    this.setState(
+      {
+        results: this.formatResults(data, newCenter),
+        markers: data.map(result => ({
+          lat: Number(result.Latitude),
+          lng: Number(result.Longitude),
+        })),
+        isLoading: false,
+        selectedLocation,
+        selectedStoreName,
+        searchResultLatlng: newCenter,
+        mapShouldPan: isLocationSelected,
+        alertType: isLocationSelected
+          ? ALERTS_TYPE.FORM_SUBMIT_SUCESS
+          : ALERTS_TYPE.WEBSITE_PURPOSE,
+      },
+      () => {
+        // remove query param from url
+        this.props.history.replace("/");
 
-        if (searchResultsContainer && searchResultsContainer.scrollIntoView) {
-          searchResultsContainer.scrollIntoView({ behavior: 'smooth' });
+        if (isLocationSelected) {
+          setTimeout(() => this.setState({ mapShouldPan: false }), 1000);
+
+          const searchResultsContainer = document.querySelector(
+            ".search-results-container"
+          );
+          if (searchResultsContainer && searchResultsContainer.scrollIntoView) {
+            searchResultsContainer.scrollIntoView();
```

Comment:
i'm trying to figure out why this is not breaking for just selecting a card. i would expect both (selecting a card and coming back after submission) to behave the same since we are just checking for isLocationSelected

Reply:
I'm confused 😅